cargo-concordium/src/build.rs

```diff
@@ -314,6 +314,45 @@ pub fn build_contract_schema<A>(
     Ok(schema)
 }
 
+/// Create a new Concordium smart contract project from a template, or there
+/// are runtime exceptions that are not expected then this function returns
+/// Err(...).
+///
+/// Otherwise a boolean is returned, signifying that the creation was
+/// successful.
+pub fn init_concordium_project(path: impl AsRef<Path>) -> anyhow::Result<bool> {
+    let path = path.as_ref();
+
+    let absolute_path = if path.is_absolute() {
+        path.to_path_buf()
+    } else {
+        env::current_dir()?.join(path)
+    };
+
+    let result = Command::new("cargo")
```

Comment:
I expect a common issue will be that users don't have `cargo-generate` installed so we should check if they do. Perhaps using the https://crates.io/crates/which crate.

Something like
```rust
if let Err(which::Error::CannotFindBinaryPath) = which::which("cargo-generate") {
   bail!("`cargo concordium init` requires `cargo-generate` which does not appear to be installed. You can install it by running `cargo install --locked cargo-generate`)
}
```
making sure that the error is displayed in a nice way.
```